### PR TITLE
Tagsegmentenhancements

### PR DESCRIPTION
**Pull request by franksmicro**
Added code to update-video-segments api to normalize tags before storing in database.  Wanted to get this out there for review as soon as possible since duplicate segments are being created daily during editing.

I also updated the schema to add segmentCount field to tags, and a one-time update script to populate this field for existing data.  However, I have not yet completed the code change to populate this field in real time, so no hurry on reviewing this.

I'm still working on the code to populate segmentCount, but expect to have ready in a few days.  The one-time script to merge existing tags will likely take quite a bit longer.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,7 +7,7 @@
   "license": "MIT",
   "scripts": {
     "start": "npm run start:local",
-    "start:local": "NODE_ENV=local nodemon index.js",
+    "start:local": "set NODE_ENV=local & nodemon index.js",
```
Comment (JKHeadley) [inline package.json:10]: I assume this is required syntax for windows. I will have to test locally to see. Most likely we will need to include both with either an OS test or different command options depending on your OS.

File: scripts/orphan-tags.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ const Manifest = require('../config/manifest.conf');
 const Config = require('../config');
 const restHapiConfig = Config.get('/restHapiConfig');
 
-require('dotenv').config();
+//require('dotenv').config();
```
Comment (JKHeadley) [inline scripts/orphan-tags.js:12]: Was `dotenv` not functioning correctly?

File: server/models/user.model.js [hunk 3]
```diff
@@ -23,29 +23,29 @@ require('firebase/firestore');
...
-    console.error('Error loading firebase admin cert:', error);
-  } else {
-    const serviceAccount = JSON.parse(data.Body.toString());
-    admin.initializeApp({
-      credential: admin.credential.cert(serviceAccount),
-      databaseURL: 'https://mbt-guide-d9b1b.firebaseio.com',
-    });
-  }
-});
+// const s3 = new AWS.S3();
```
Comment (JKHeadley) [inline server/models/user.model.js:26]: We won't want to merge the commented code to master.

File: server/models/user.model.js [hunk 4]
```diff
@@ -287,31 +287,31 @@ module.exports = function(mongoose) {
...
-          Log.error(err);
-          return false;
-        }
+        // try {
+        //   result = await firebaseApp.auth().signInWithEmailAndPassword(email, password);
+        // } catch (err) {
+        //   //TODO: If test user, check if error is because user doesn't exist and if so, create the user (in firebase)
+        //   Log.error(err);
+        //   return false;
+        // }
```
Comment (JKHeadley) [inline server/models/user.model.js:296]: Same here.

File: server/models/user.model.js [hunk 5]
```diff
@@ -287,31 +287,31 @@ module.exports = function(mongoose) {
...
+          email: email.toLowerCase(),
           isDeleted: false,
         };
 
         let mongooseQuery = self.findOne(query);
 
         let user = await mongooseQuery.lean();
 
-        user = await this.updateActive({ firebaseUser: result.user, user });
+        //user = await this.updateActive({ firebaseUser: result.user, user });
```
Comment (JKHeadley) [inline server/models/user.model.js:307]: Same.

File: server/models/user.model.js [hunk 6]
```diff
@@ -287,31 +287,31 @@ module.exports = function(mongoose) {
...
+        //   result = await firebaseApp.auth().signInWithEmailAndPassword(email, password);
+        // } catch (err) {
+        //   //TODO: If test user, check if error is because user doesn't exist and if so, create the user (in firebase)
+        //   Log.error(err);
+        //   return false;
+        // }
 
         const query = {
-          email: result.user.email.toLowerCase(),
+          email: email.toLowerCase(),
```
Comment (JKHeadley) [inline server/models/user.model.js:299]: I think this is also part of the local changes to your system. If so then let's not merge this.

File: utilities/querytest.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+'use strict';
```
Comment (JKHeadley) [inline utilities/querytest.js:1]: Could you add a description to what you use this test for?

File: utilities/querytest_save.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
+'use strict';
```
Comment (JKHeadley) [inline utilities/querytest_save.js:1]: Same as above, could you add a description for this?

**Review by JKHeadley — CHANGES_REQUESTED** [review]

**Comment by franksmicro** [comment]
I accidentally pushed some changes that were only meant for my local machine.

The changes requested in this PR have all been implemented as part of other PRs, so this PR is now obsolete.

**Pull request closed without merging**